bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/handler/SmartHomeDeviceHandler.java

```diff
@@ -255,11 +252,7 @@ public void updateChannelStates(List<SmartHomeBaseDevice> allDevices,
             }
         }
 
-        if (stateFound) {
-            updateStatus(ThingStatus.ONLINE);
```

Comment:
maybe we a debug trace instead of the updateStatus would be better. Removing this "if" complete would make diagnostic much harder.

Reply:
The logging in l. 228 is exactly that. It just exists early and skips the processing the lookup of applianceIds and interfaces if the state array is empty.